src/librustc_mir/interpret/memory.rs

```diff
@@ -847,6 +847,12 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         let dest_bytes =
             self.get_raw_mut(dest.alloc_id)?.get_bytes_mut(&tcx, dest, size * length)?.as_mut_ptr();
 
+        // If `dest_bytes` is empty we just optimize to not run anything for zsts.
+        // See #67539
+        if dest_bytes.is_empty() {
```

Comment:
I guess you need to split `dest_bytes` into the slice part, then do the early abort, then do the `as_mut_ptr`

Reply:
Yeah, was to fast to push :)